src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -68,9 +36,28 @@ static string GetRoutingKey(Type type, string key = "")
             return key + type.FullName.Replace(".", "-");
         }
 
-        static readonly byte[] MsPublicKeyToken = typeof(string).Assembly.GetName().GetPublicKeyToken();
+        static bool IsSystemType(Type type)
+        {
+            bool result;
+
+            if (!IsSystemTypeCache.TryGetValue(type, out result))
+            {
+                var nameOfContainingAssembly = type.Assembly.GetName().GetPublicKeyToken();
+                IsSystemTypeCache[type] = result = IsClrType(nameOfContainingAssembly);
+            }
+
+            return result;
+        }
+
+        static bool IsClrType(byte[] a1)
+        {
+            IStructuralEquatable structuralEquatable = a1;
+            return structuralEquatable.Equals(MsPublicKeyToken, StructuralComparisons.StructuralEqualityComparer);
+        }
+
+        static bool IsNServiceBusMarkerInterface(Type type) => type == typeof(IMessage) || type == typeof(ICommand) || type == typeof(IEvent);
```

Comment:
The problem with that is that everyone's VS is different, but everyone's GitHub is the same.

Anyway, we can continue this (hugely compelling) discussion elsewhere. :smile: